### ISSUE TITLE
Populate file.uri on 0th file element if it doesn't match the key

### ISSUE THREAD

**Issue by ghost** [sarif-standard/sarif-spec#160]
Per @michaelcfanning:

Consider this nested thing ModernCop reports. In order to get the root URI form the 0th element, you have to parse the key of the dictionary. That’s not good. If the URI of the 0th item doesn’t match, it makes most sense for producers to break out the root uri. This means that pathFromParent is basically a path from a null parent though. If you know what I mean.

``` json
"file:///d:/src/SecDevTools/Main/src/ModernCop/Source/Mod50.Test.Unit/ModernCopTestData/Mod50/CompiledPackages/SimpleJSApp.appxbundle#/SimpleJSApp_1.0.0.1_AnyCPU.appx/pages/section/section.html": [
  {
    "mimeType": "application/octet-stream"
  },
  {
    "pathFromParent": "/SimpleJSApp_1.0.0.1_AnyCPU.appx",
    "mimeType": "application/vns.ms-appx"
  },
  {
    "pathFromParent": "/pages/section/section.html",
    "mimeType": "text/html"
  }
]
```

FIXED?

``` json
"file:///d:/src/SecDevTools/Main/src/ModernCop/Source/Mod50.Test.Unit/ModernCopTestData/Mod50/CompiledPackages/SimpleJSApp.appxbundle#/SimpleJSApp_1.0.0.1_AnyCPU.appx/pages/section/section.html": [
  {
    "pathFromParent": " file:///d:/src/SecDevTools/Main/src/ModernCop/Source/Mod50.Test.Unit/ModernCopTestData/Mod50/CompiledPackages/SimpleJSApp.appxbundle",
    "mimeType": "application/octet-stream"
  },
  {
    "pathFromParent": "/SimpleJSApp_1.0.0.1_AnyCPU.appx",
    "mimeType": "application/vns.ms-appx"
  },
  {
    "pathFromParent": "/pages/section/section.html",
    "mimeType": "text/html"
  }
]
```

**Comment by ghost** [comment]
After consideration:

Since we’ve now stated that the key is not a URI – and so we can’t guarantee that it can be parsed as a URI, such as `file:///c:/foo/bar.zip#/baz/qux#xyzzy`– then we can’t ask consumers to feed it into their “Uri” class and strip off the fragment. So we need to do this.

**Comment by ghost** [comment]
Ok, `pathFromParent` _is_ once again a URI and has been renamed to `uri`. But the text as given in Microsoft/sarif-sdk#196 isn't quite right. It doesn't say that you _can_ omit `uri` on the first element of the `file` array, in which case consumers are expected to figure it out from the non-fragment portion of the property name -- which is, as I said, now once again guaranteed to be a valid URI. Come to think of it, the text doesn't say that, either.

I'll fix this up separately once Microsoft/sarif-sdk#196 is merged.

@michaelcfanning FYI

**Comment by ghost** [comment]
Fixed by 8495747. @michaelcfanning FYI
